core/src/main/java/com/squarespace/template/plugins/platform/CommerceFormatters.java

```diff
@@ -466,18 +468,67 @@ private static String defaultSubscriptionPriceString(JsonNode productNode) {
 
       return sb.toString();
     }
+  }
 
-    private static String getMoneyString(JsonNode moneyNode, Context ctx) {
-      if (CommerceUtils.useCLDRMode(ctx)) {
-        Decimal amount = CommerceUtils.getAmountFromMoneyNode(moneyNode);
-        String currencyCode = CommerceUtils.getCurrencyFromMoneyNode(moneyNode);
-        return PluginUtils.formatMoney(amount, currencyCode, ctx.cldr());
-      } else {
-        Decimal legacyAmount = CommerceUtils.getLegacyPriceFromMoneyNode(moneyNode);
-        StringBuilder buf = new StringBuilder();
-        CommerceUtils.writeLegacyMoneyString(legacyAmount, buf);
-        return buf.toString();
+  protected static class SubscriptionPriceFormatter extends BaseFormatter {
+
+    private Instruction template;
+
+    public SubscriptionPriceFormatter() {
+      super("subscription-price", false);
+    }
+
+    @Override
+    public void initialize(Compiler compiler) throws CodeException {
+      String source = loadResource(CommerceFormatters.class, "subscription-price.html");
+      this.template = compiler.compile(source.trim()).code();
+    }
+
+    @Override
+    public void apply(Context ctx, Arguments args, Variables variables) throws CodeExecuteException {
+      Variable var = variables.first();
+      String subscriptionId = args.isEmpty() ? null : args.first();
+      JsonNode node = var.node();
+      StringBuilder buffer = new StringBuilder();
+      ObjectNode subscriptionResults = JsonUtils.createObjectNode();
+
+      JsonNode pricingOptions = CommerceUtils.getPricingOptionsAmongLowestVariant(node);
```

Comment:
We need to get the list of `pricingOptions` for the lowest cost variant taking into account `onSale` prices.